Firestore/core/src/firebase/firestore/core/query.cc

```diff
@@ -27,6 +31,27 @@ using model::Document;
 using model::DocumentKey;
 using model::ResourcePath;
 
+namespace {
+
+// Convert a vector of unique_ptr's to a vector of shared_ptr's
+std::vector<std::shared_ptr<core::Filter>> Convert(
+    std::vector<std::unique_ptr<core::Filter>>&& v) {
+  std::vector<std::shared_ptr<core::Filter>> result;
+  std::for_each(v.begin(), v.end(), [&](std::unique_ptr<core::Filter>& f) {
+    result.push_back(std::move(f));
+  });
+  v.clear();
```

Comment:
I'm slightly leaning towards omitting it, because like you said, the function is small and has a well-defined task. I don't feel strongly either way, though -- I'm fine if you keep it, too.